### PR TITLE
Feature/convert bolt list to web component

### PR DESCRIPTION
**Pull request by adam2661**
## Jira

http://vjira2:8080/browse/BDS-761

## Summary

Convert bolt list component to web component

## Details

Retype JS and SCSS to work with Web Components

## How to test

Run the branch locally and check pages under Components->List in the navigation.

### CODE REVIEW

**Comment by adam2661** [comment]
@mikemai2awesome  and @sghoweri i want to resolve this conflict and my question is: that i do the local marge with master and then push to this PR or the conflicts are resolved in a different way?

**Comment by sghoweri** [comment]
> @mikemai2awesome and @sghoweri i want to resolve this conflict and my question is: that i do the local marge with master and then push to this PR or the conflicts are resolved in a different way?

@adam2661 merging the latest on master is always good to resolve the merge conflicts + also make sure feature branches don't diverge too much (which helps with testing and releasing)

**Comment by remydenton** [comment]
Sorry, I should have posted in the main Slack channel to have us all on the same page.  @adam2661 and I discussed 1-on-1 yesterday and I advised him to merge in this case, as we commonly do.

FWIW, rebasing against `master` would actually have been the cleanest way, but @adam2661 had just rebased against `feature/lit-html-renderer` as I'd suggested before I realized that `feature/lit-html-renderer` had already been merged to `master`.

In summary for everyone:
- Practice rebasing when you can to get comfortable with it!
- Rebase rather than merge when you have not yet created a PR
- Merge rather than rebase after you've created a PR to avoid rewriting history (as in this case)

**Comment by adam2661** [comment]
Hi, @mikemai2awesome and @sghoweri i think i cover are your requests. Can you guys check if everything is ok because i'm don't know if my implementation of passing `bolt-list-item` is correct and efficient?
